src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -372,20 +625,59 @@ public bool TryGet(ref SequencePosition position, out ReadOnlyMemory<T> memory,
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private ReadOnlySequence<T> SliceImpl(in SequencePosition begin, in SequencePosition end)
+        private ReadOnlySequence<T> SliceImpl(in SequencePosition start, in SequencePosition end)
         {
             // In this method we reset high order bits from indices
             // of positions that were passed in
             // and apply type bits specific for current ReadOnlySequence type
 
             return new ReadOnlySequence<T>(
-                begin.GetObject(),
-                begin.GetInteger() & ReadOnlySequence.IndexBitMask | (Start.GetInteger() & ReadOnlySequence.FlagBitMask),
+                start.GetObject(),
+                start.GetInteger() & ReadOnlySequence.IndexBitMask | (_sequenceStart.GetInteger() & ReadOnlySequence.FlagBitMask),
                 end.GetObject(),
-                end.GetInteger() & ReadOnlySequence.IndexBitMask | (End.GetInteger() & ReadOnlySequence.FlagBitMask)
+                end.GetInteger() & ReadOnlySequence.IndexBitMask | (_sequenceEnd.GetInteger() & ReadOnlySequence.FlagBitMask)
             );
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private ReadOnlySequence<T> SliceImpl(object beginObject, int beginInteger, object endObject, int endInteger)
+        {
+            // In this method we apply type bits specific for current ReadOnlySequence type
+
+            return new ReadOnlySequence<T>(
+                   beginObject,
+                   beginInteger | (_sequenceStart.GetInteger() & ReadOnlySequence.FlagBitMask),
+                   endObject,
+                   endInteger | (_sequenceEnd.GetInteger() & ReadOnlySequence.FlagBitMask)
+               );
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private ReadOnlySequence<T> SliceImpl(in SequencePosition start)
+        {
+            // In this method we apply type bits specific for current ReadOnlySequence type
+
+            return new ReadOnlySequence<T>(
+                   start.GetObject(),
+                   start.GetInteger() & ReadOnlySequence.IndexBitMask | (_sequenceStart.GetInteger() & ReadOnlySequence.FlagBitMask),
```

Comment:
start.GetInteger() & ReadOnlySequence.IndexBitMask => GetInteger(start) can be used